subsys/net/l2/wifi/wifi_mgmt.c

```diff
@@ -157,3 +164,71 @@ static int wifi_ap_disable(uint32_t mgmt_request, struct net_if *iface,
 }
 
 NET_MGMT_REGISTER_REQUEST_HANDLER(NET_REQUEST_WIFI_AP_DISABLE, wifi_ap_disable);
+
+static int wifi_iface_status(uint32_t mgmt_request, struct net_if *iface,
+			  void *data, size_t len)
+{
+	int ret;
+	const struct device *dev = net_if_get_device(iface);
+	struct net_wifi_mgmt_offload *off_api =
+		(struct net_wifi_mgmt_offload *) dev->api;
+	struct wifi_iface_status status = { 0 };
+
+	if (off_api == NULL || off_api->iface_status == NULL) {
+		return -ENOTSUP;
+	}
+
+	if (!data || len != sizeof(struct wifi_iface_status)) {
+		return -EINVAL;
+	}
+
+	ret = off_api->iface_status(dev, &status);
+
+	if (ret) {
+		return ret;
+	}
+
+	memcpy(data, &status, len);
+
+	return 0;
+}
+NET_MGMT_REGISTER_REQUEST_HANDLER(NET_REQUEST_WIFI_IFACE_STATUS, wifi_iface_status);
+
+void wifi_mgmt_raise_iface_status_event(struct net_if *iface,
+		struct wifi_iface_status *iface_status)
```

Comment:
I am not sure this is needed. First you define `NET_MGMT_REGISTER_REQUEST_HANDLER(NET_REQUEST_WIFI_IFACE_STATUS, wifi_iface_status);` with implementation that looks synchronous. Then you add `wifi_mgmt_raise_iface_status_event`, which is (I think) asynchronous response to request. I guess we should go with synchronous or asynchronous, not both. See "connect" and "disconnect" which are asynchronous requests. See "ap_enable" which is synchronous request.

Reply:
We can support both, e.g., if an application wants to fetch status then it can use the `synchronous` way, and if the underlying driver/wpa_supplicant wants to notify the application of an important event then it can use `asynchronous` way.

Reply:
Oh, I get what you mean. This will be the first wifi_mgmt/net_mgmt API that does this, right?

Reply:
Yes.